### PR TITLE
ACM-2666 Reflect CP upgrade in UI right away + unit tests

### PR DESCRIPTION
**Pull request by o-farag**
Signed-off-by: Omar Farag <ofarag@redhat.com>

Issue: https://issues.redhat.com/browse/ACM-2666

Peek 2023-01-23 13-28

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx [hunk 1]
```diff
@@ -15,18 +15,24 @@ import {
...
 import { RecoilRoot } from 'recoil'
 import { ansibleJobState, clusterImageSetsState, nodePoolsState } from '../../../../../atoms'
 import { nockIgnoreApiPaths, nockIgnoreRBAC, nockRBAC } from '../../../../../lib/nock-util'
 import { clickByText, waitForCalled, waitForNock, waitForNotText, waitForText } from '../../../../../lib/test-util'
 import { DistributionField } from './DistributionField'
 import { Router } from 'react-router-dom'
+import { HostedClusterK8sResource } from 'openshift-assisted-ui-lib/cim'
+import userEvent from '@testing-library/user-event'
+
+import '@testing-library/jest-dom'
```
Comment (chenz4027) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx:35]: Ideally this shouldn't be needed as it's defined in the frontend/src/setupTests.ts, but could be wrong here. If it's required, just ignore my comment ;)

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx [hunk 2]
```diff
@@ -934,4 +1032,171 @@ describe('DistributionField hypershift clusters', () => {
...
+      mockHypershiftCluster,
+      true,
+      false,
+      undefined,
+      undefined,
+      mockHostedCluster,
+      false
+    )
+
+    //await new Promise((resolve) => setTimeout(resolve, 500)) // makes sure everything is finished
```
Comment (chenz4027) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx:1192]: I think you forgot to remove the comment and debug line.

**Review by chenz4027 — CHANGES_REQUESTED** [review]
Great work Omar, it's difficult to hit the coverage and you did great 👍

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx (hunk 1 above)
Comment (o-farag) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx]: You're correct, it wasn't needed :)

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx (hunk 2 above)
Comment (o-farag) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx]: Removed

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx (hunk 1 above)
Comment (chenz4027) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx]: Nice, thanks Omar!

**Review by chenz4027 — APPROVED** [review]
/lgtm

**Pull request merged**